src/adapter/src/coord/sequencer.rs

```diff
@@ -105,430 +105,442 @@ impl Coordinator {
             return ctx.retire(Err(e));
         }
 
-        match plan {
-            Plan::CreateSource(plan) => {
-                let source_id = return_if_err!(self.catalog_mut().allocate_user_id().await, ctx);
-                let result = self
-                    .sequence_create_source(
-                        ctx.session_mut(),
-                        vec![CreateSourcePlans {
-                            source_id,
-                            plan,
-                            resolved_ids,
-                        }],
-                    )
-                    .await;
-                ctx.retire(result);
-            }
-            Plan::CreateSources(plans) => {
-                assert!(
-                    resolved_ids.0.is_empty(),
-                    "each plan has separate resolved_ids"
-                );
-                let result = self.sequence_create_source(ctx.session_mut(), plans).await;
-                ctx.retire(result);
-            }
-            Plan::CreateConnection(plan) => {
-                self.sequence_create_connection(ctx, plan, resolved_ids)
-                    .await;
-            }
-            Plan::CreateDatabase(plan) => {
-                let result = self.sequence_create_database(ctx.session_mut(), plan).await;
-                ctx.retire(result);
-            }
-            Plan::CreateSchema(plan) => {
-                let result = self.sequence_create_schema(ctx.session_mut(), plan).await;
-                ctx.retire(result);
-            }
-            Plan::CreateRole(plan) => {
-                let result = self.sequence_create_role(ctx.session(), plan).await;
-                if result.is_ok() {
-                    self.maybe_send_rbac_notice(ctx.session());
+        loop {
```

Comment:
I don't think we should `loop` here, especially since it's just for a single case. Above this `match` could we instead do something like:

```
if let Plan::ShowColumns(...) = plan {
  plan = ...
  resolved_ids = ...
}